optaplanner-core/src/test/java/org/optaplanner/core/api/score/stream/IntervalJoinersTest.java

```diff
@@ -0,0 +1,951 @@
+                (AbstractBiJoiner<Interval<Long, BigInteger>, Interval<Long, BigDecimal>>) Joiners.overlaps(leftStartMapping,
+                        leftEndMapping, rightStartMapping, rightEndMapping);
+
+        assertSoftly(softly -> {
+            // True cases (equals, overlaps, contains, starts, ends)
+            softly.assertThat(joiner.matches(Interval.ofBigInt(1L, 5L),
+                    Interval.ofBigDecimal(1L, 5L)))
+                    .as("Case A = B").isTrue();
+
+            softly.assertThat(joiner.matches(Interval.ofBigInt(3L, 7L),
+                    Interval.ofBigDecimal(1L, 5L)))
+                    .as("B starts before A, A ends after B").isTrue();
+
+            softly.assertThat(joiner.matches(Interval.ofBigInt(1L, 5L),
+                    Interval.ofBigDecimal(3L, 7L)))
+                    .as("A starts before B, B ends after A").isTrue();
+
+            softly.assertThat(joiner.matches(Interval.ofBigInt(3L, 5L),
+                    Interval.ofBigDecimal(1L, 7L)))
+                    .as("B contains A").isTrue();
+
+            softly.assertThat(joiner.matches(Interval.ofBigInt(1L, 7L),
+                    Interval.ofBigDecimal(3L, 5L)))
+                    .as("A contains B").isTrue();
+
+            softly.assertThat(joiner.matches(Interval.ofBigInt(1L, 7L),
+                    Interval.ofBigDecimal(1L, 3L)))
+                    .as("A started by B").isTrue();
+
+            softly.assertThat(joiner.matches(Interval.ofBigInt(1L, 3L),
+                    Interval.ofBigDecimal(1L, 7L)))
+                    .as("B started by A").isTrue();
+
+            softly.assertThat(joiner.matches(Interval.ofBigInt(1L, 7L),
+                    Interval.ofBigDecimal(3L, 7L)))
+                    .as("A ended by B").isTrue();
+
+            softly.assertThat(joiner.matches(Interval.ofBigInt(3L, 7L),
+                    Interval.ofBigDecimal(1L, 7L)))
+                    .as("B ended by A").isTrue();
+
+            // False Cases (before, after, meets)
+
+            softly.assertThat(joiner.matches(Interval.ofBigInt(1L, 3L),
+                    Interval.ofBigDecimal(5L, 7L)))
+                    .as("A before B").isFalse();
+
+            softly.assertThat(joiner.matches(Interval.ofBigInt(5L, 7L),
+                    Interval.ofBigDecimal(1L, 3L)))
+                    .as("B before A").isFalse();
+
+            // This is false since typically, when overlaps is used,
+            // end is exclusive, and start is inclusive,
+            // so 0-5, 5-10 do not overlap
+            softly.assertThat(joiner.matches(Interval.ofBigInt(1L, 3L),
+                    Interval.ofBigDecimal(3L, 7L)))
+                    .as("A meets B").isFalse();
+
+            softly.assertThat(joiner.matches(Interval.ofBigInt(3L, 7L),
+                    Interval.ofBigDecimal(1L, 3L)))
+                    .as("B meets A").isFalse();
+        });
+    }
+
+    @Test
+    public void overlapsBiShort() {
```

Comment:
How does overlapsBiShort() improve the test coverage over overlapsBi?

Reply:
It for the short form of the `overlapping` method for the same type (i.e. `overlapping(Function, Function)` NOT `overlapping(Function, Function, Function, Function)`)

Reply:
Changed to `overlapsDifferentTypes` and `overlapsSameTypes`